contracts/core/Dispatcher.sol

```diff
@@ -86,11 +87,20 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
         L1Header calldata l1header,
         OpL2StateProof calldata proof,
         uint256 height,
-        uint256 appHash
+        uint256 appHash,
+        uint256 clientId
     ) external returns (uint256 fraudProofEndTime, bool ended) {
+        LightClient lightClient = _clientIdToLightClient[clientId];
+        if (address(lightClient) == address(0)) {
+            revert IBCErrors.lightClientNotFound(clientId);
+        }
         return lightClient.addOpConsensusState(l1header, proof, height, appHash);
     }
 
+    function addNewConnection(string calldata connection, LightClient lightClient) external onlyOwner {
+        _addNewConnection(connection, lightClient);
+    }
```

Comment:
can we change `addNewConnection` to `setClientForConnection` (or add as a new method) so it supports both adding new and updating existing `connectionID -> client` map?
if client == address(0), then a connection must be removed/disabled 
we might need to update existing connections' clients in the future

Reply:
yes I think this is valid, should be changed now. we can have setting the client to address 0 as a seperate method

Reply:
tests should be added in [this commit](https://github.com/open-ibc/vibc-core-smart-contracts/pull/70/commits/062e7534ba0d92482f489cfc6deba827cbd47148)!

Reply:
err now [this commit ](https://github.com/open-ibc/vibc-core-smart-contracts/pull/70/commits/77a4e5ea417b940447ab81c1bec44af1cd18dac3)after rebasing